projects/collection-editor-library/src/lib/services/editor/editor.service.ts

```diff
@@ -530,21 +532,37 @@ export class EditorService {
 
   getBranchingLogicByFolder(identifier) {
     const nodeData = this.treeService.getNodeById(identifier);
+    this.selectedPrimaryCategory=_.get(nodeData,'data.primaryCategory');
     const branchingLogic = _.get(nodeData, 'data.metadata.branchingLogic');
     return branchingLogic || {};
   }
 
   getDependentNodes(identifier) {
+    const parentBranchingLogic=this.getParentIdentifier(identifier);
+    if (!_.isEmpty(parentBranchingLogic)) {
+     const branchingEntry = this.getBranchingLogicEntry(parentBranchingLogic,identifier);
+     if(!_.isEmpty(branchingEntry.source[0])) {
+       const parentBranchingLogic=this.getParentIdentifier(branchingEntry.source[0]);
+       const branchingEntrys =this.getBranchingLogicEntry(parentBranchingLogic,branchingEntry.source[0]);
+       return !_.isEmpty(branchingEntrys) ? { source:branchingEntry.source, target: branchingEntrys.target } :{};
+     }else{
+      return !_.isEmpty(branchingEntry) ? { source: branchingEntry.source, target: branchingEntry.target } :{};
+     }
+    }
+  }
+
+  getParentIdentifier(identifier){
     const leafNode = this.treeService.getNodeById(identifier);
     const parentIdentifier = _.get(leafNode, 'data.metadata.parent');
     const parentBranchingLogic = this.getBranchingLogicByFolder(parentIdentifier);
-    if (!_.isEmpty(parentBranchingLogic)) {
-     const branchingEntry = _.find(parentBranchingLogic, (logic, key) => {
-        return key === identifier;
-      });
-     return !_.isEmpty(branchingEntry) ? { source: branchingEntry.source, target: branchingEntry.target } :
-     {};
-    }
+    return parentBranchingLogic;
```

Comment:
is this method returning parent Identifier or parent branching logic? try to keep the method name relevant to its purpose